src/app/sales/sale.class.ts

```diff
@@ -0,0 +1,33 @@
+import { ICake } from "../cakes/cake.class";
+
+export interface ISale {
+    customerName: string,
+    customerPhoneNumber: string,
+    customerEmail: string,
+    totalAmount: number,
+    cake: SoldCake,
+}
+
+export class SoldCake implements ICake {
```

Comment:
Since we are using a Single File Responsibility file distribution, this class should live in a separate file named `sold-cake.class.ts`.